modules/cudf/src/data_frame.ts

```diff
@@ -450,6 +450,71 @@ export class DataFrame<T extends TypeMap = any> {
     return new DataFrame(series_map);
   }
 
+  /**
+   * Returns the first n rows as a new DataFrame.
+   *
+   * @param n The number of rows to return.
+   *
+   * @example
+   * ```typescript
+   * import {DataFrame, Series, Int32} from '@rapidsai/cudf';
+   *
+   * const df = new DataFrame({
+   *   a: Series.new({type: new Int32, data: [0, 1, 2, 3, 4, 5, 6]}),
+   *   b: Series.new([0.0, 1.0, 2.0, 3.0, 4.0, 5.0, 6.0])
+   * });
+   *
+   * a.head();
+   * // {a: [0, 1, 2, 3, 4], b: [0.0, 1.0, 2.0, 3.0, 4.0]}
+   *
+   * b.head(1);
+   * // {a: [0], b: [0.0]}
+   *
+   * a.head(-1);
+   * // throws index out of bounds error
+   * ```
+   */
+  head(n = 5): DataFrame {
```

Comment:
Need to preserve the generic type signature:

```suggestion
  head(n = 5): DataFrame<T> {
```